File: skills/pz-director/scripts/greeting_trigger.py
#!/usr/bin/env python3
"""Fail-closed trigger gate for the SIMON Greeting Dispatcher cron."""
import json
import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).parent))
from simon_online_gate import check_players_online

QUEUE_FILE = Path(__file__).resolve().parent.parent / "state" / "greeting-queue.json"


def emit(fire: bool, reason: str) -> None:
    print(json.dumps({"fire": fire, "reason": reason}, separators=(",", ":")))
    sys.exit(0)


def main() -> None:
    # A stale queue must never wake the LLM after everyone has disconnected.
    presence = check_players_online()
    if not presence.get("online"):
        emit(False, "offline")

    try:
        queue = json.loads(QUEUE_FILE.read_text()) if QUEUE_FILE.exists() else {}
    except (json.JSONDecodeError, OSError):
        emit(False, "queue-unreadable")

    pending = queue.get("pending", [])
    if isinstance(pending, list) and pending:
        emit(True, "pending-and-online")

    emit(False, "no-pending-greeting")


if __name__ == "__main__":
    main()

File: skills/pz-director/scripts/greeting_trigger.js
#!/usr/bin/env node
/** Compatibility wrapper: all scheduled LLM gating lives in Python. */
const path = require('path');
const { spawnSync } = require('child_process');

const script = path.join(__dirname, 'greeting_trigger.py');
const result = spawnSync('python3', [script], { encoding: 'utf8' });

if (result.error || result.status !== 0 || !result.stdout) {
  process.stdout.write(JSON.stringify({ fire: false, reason: 'python-gate-failed' }) + '\n');
  process.exit(0);
}

process.stdout.write(result.stdout.trim() + '\n');
process.exit(0);
